pkg/agent/run.go

```diff
@@ -115,7 +116,6 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	}
 
 	notifySocket := os.Getenv("NOTIFY_SOCKET")
-	os.Unsetenv("NOTIFY_SOCKET")
```

Comment:
IIRC we need to leave this around so that other subprocesses forked off after this (kubelet and containerd) don't also try to notify.

Reply:
It's been so long now but I believe you're correct.

Reply:
Fixed!